### PR TITLE
Airneis 35 : Vault Client

### PR DESCRIPTION
**Pull request by cedricgautier**
This client will enable us to fetch secrets safely!

### CODE REVIEW

**Review by KyDev25 — APPROVED** [review]
Good for me!

File: back/lib/shared/vault/vault_test.go [hunk 1]
```diff
@@ -3,12 +3,41 @@ package vault_test
...
+		})
+
+		secret, err := vlt.ReadSecret("hello_world_secret")
+		It("should return the secret value without error", func() {
+			Expect(err).To(Not(HaveOccurred()))
+			Expect(secret).To(Equal("hello_world"))
+		})
+	})
+})
+*/
```
Comment (barnabepilliaudin) [inline back/lib/shared/vault/vault_test.go]: Shouldn't this code be removed?

**Review by BarnabePILLIAUDIN — COMMENTED** [review]
Should the dead code be removed?

File: back/lib/shared/vault/vault_test.go (hunk 1 above)
Comment (cedricgautier) [reply back/lib/shared/vault/vault_test.go]: It's going to be used later on, I'll remove it for now. THanks

**Review by BarnabePILLIAUDIN — APPROVED** [review]
LGTM!

File: .gitignore [hunk 2]
```diff
@@ -53,4 +53,8 @@ next-env.d.ts
 bun.lockb
 
 test_report*
-.env
\ No newline at end of file
+.env
+
+
```
Comment (avetisk) [inline .gitignore:58]: Too many blank lines 👀

File: back/lib/shared/vault/vault_test.go [hunk 3]
```diff
@@ -3,12 +3,11 @@ package vault_test
...
-	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
-func TestVaultAPI(t *testing.T) {
+func TestVaultClient(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecsWithDefaultAndCustomReporters(t, "vault", []Reporter{reporters.NewJUnitReporter("test_report_vault.xml")})
+	RunSpecs(t, "VaultClient")
```
Comment (avetisk) [inline back/lib/shared/vault/vault_test.go:12]: That's all your test?? 👀

**Pull request merged**